tidy3d/plugins/smatrix/ports/rectangular_lumped.py

```diff
@@ -0,0 +1,265 @@
+
+    @cached_property
+    def injection_axis(self):
+        """Injection axis of the port."""
+        return self.size.index(0.0)
+
+    @pd.validator("voltage_axis", always=True)
+    def _voltage_axis_in_plane(cls, val, values):
+        """Ensure voltage integration axis is in the port's plane."""
+        size = values.get("size")
+        if val == size.index(0.0):
+            raise ValidationError("'voltage_axis' must lie in the port's plane.")
+        return val
+
+    @cached_property
+    def current_axis(self) -> Axis:
+        """Integration axis for computing the port current via the magnetic field."""
+        return 3 - self.injection_axis - self.voltage_axis
+
+    def to_source(
+        self, source_time: GaussianPulse, snap_center: float, grid: Grid
+    ) -> UniformCurrentSource:
+        """Create a current source from the lumped port."""
+        # Discretized source amps are manually zeroed out later if they
+        # fall on Yee grid locations outside the analytical source region.
+        component = "xyz"[self.voltage_axis]
+        center = list(self.center)
+        if snap_center:
+            center[self.injection_axis] = snap_center
+        return UniformCurrentSource(
+            center=center,
+            size=self.size,
+            source_time=source_time,
+            polarization=f"E{component}",
+            name=self.name,
+            interpolate=True,
+            confine_to_bounds=True,
+        )
+
+    def to_load(self, snap_center: float) -> LumpedResistor:
+        """Create a load resistor from the lumped port."""
+        # 2D materials are currently snapped to the grid, so snapping here is not needed.
+        # It is done here so plots of the simulation will more accurately portray the setup
+        center = list(self.center)
+        if snap_center:
+            center[self.injection_axis] = snap_center
+        return LumpedResistor(
+            center=center,
+            size=self.size,
+            num_grid_cells=self.num_grid_cells,
+            resistance=np.real(self.impedance),
+            name=f"{self.name}_resistor",
+            voltage_axis=self.voltage_axis,
+        )
+
+    def to_voltage_monitor(self, freqs: FreqArray, snap_center: float) -> FieldMonitor:
+        """Field monitor to compute port voltage."""
+        center = list(self.center)
+        if snap_center:
+            center[self.injection_axis] = snap_center
+
+        e_component = "xyz"[self.voltage_axis]
+        # Size of voltage monitor can essentially be 1D from ground to signal conductor
+        voltage_mon_size = list(self.size)
+        voltage_mon_size[self.injection_axis] = 0.0
+        voltage_mon_size[self.current_axis] = 0.0
+        # Create a voltage monitor
+        return FieldMonitor(
+            center=center,
+            size=voltage_mon_size,
+            freqs=freqs,
+            fields=[f"E{e_component}"],
+            name=f"{self.name}_voltage",
```

Comment:
would be nice for this to be a method 

```py
def _voltage_monitor_name(self) -> str:
    return f"{self.name}_voltage"
```

so we could use it here and below. and then if it changes, we just need to change it once.

same for current